File: go/sublist/sublist.go
package sublist

type Relation string

const (
    equal Relation = "equal"
    sublist Relation = "sublist"
    superlist Relation = "superlist"
    unequal Relation = "unequal"
)

func Sublist(l1, l2 []int) Relation {
    sublistBool := false
    superlistBool := false
    if len(l1) == 0 {
        sublistBool = true
    }
	if len(l2) == 0 {
        superlistBool = true
    }
	if len(l1) <= len(l2) {
        for i := 0; i <= len(l2) - len(l1); i++ {
            for j := 0; j < len(l1); j++ {
                if l1[j] != l2[i + j] {
                    break
                }
            	if j == len(l1) - 1 {
                    sublistBool = true
                }
            }
        }
    }
    if len(l1) >= len(l2) {
        for i := 0; i <= len(l1) - len(l2); i++ {
            for j := 0; j < len(l2); j++ {
                if l2[j] != l1[i + j] {
                    break
                }
            	if j == len(l2) - 1 {
                    superlistBool = true
                }
            }
        }
    }
	switch {
    case sublistBool && superlistBool:
    	return equal
    case sublistBool && !superlistBool:
    	return sublist
    case !sublistBool && superlistBool:
    	return superlist
    default:
    	return unequal
    }
}

File: python/sublist/sublist.py
"""
This exercise stub and the test suite contain several enumerated constants.

Since Python 2 does not have the enum module, the idiomatic way to write
enumerated constants has traditionally been a NAME assigned to an arbitrary,
but unique value. An integer is traditionally used because it’s memory
efficient.
It is a common practice to export both constants and functions that work with
those constants (ex. the constants in the os, subprocess and re modules).

You can learn more here: https://en.wikipedia.org/wiki/Enumerated_type
"""

SUBLIST = 0
SUPERLIST = 1
EQUAL = 2
UNEQUAL = 3


def sublist(list_one, list_two):
    if list_one == list_two:
        return EQUAL
    if contains(list_two, list_one):
        return SUBLIST
    if contains(list_one, list_two):
        return SUPERLIST
    return UNEQUAL


def contains(list_one, list_two):
    for i in range(len(list_one) - len(list_two) + 1):
        if list_one[slice(i, i + len(list_two))] == list_two:
            return True
    return False
